### DISCUSSION TITLE
How to perform non-FAST processing on streamer data?

### DISCUSSION THREAD

**Discussion by prash-p** [FAST-Imaging/FAST#191] [Q&A] [answered]
Hi, 
I want to use the ClariusStreamer and OpenIGTLinkStreamer classes to stream data and I would like to perform image processing on the resulting data for an indefinite period of time. I want to use tools like numpy and PIL instead of the FAST filters. Could you share example code for doing this?
Thanks!

**Comment by smistad** [comment, marked as answer]
Hi. There is an example of how to do that here https://fast.eriksmistad.no/python_process_object_8py-example.html

Just swap the ImageFileStreamer with the OpenIGTLinkStreamer or ClariusStreamer

**Reply by prash-p** [reply]
What are the limits of the Python process object class? Can I for example wrap it around a pytorch neural network to do inference on the stream?

**Reply by andreped** [reply]
That should work fine, @prash-p. But I would recommend converting the pytorch model to ONNX and using the Neural network PO instead for faster inference, for instance using the TensorRT inference engine.

**Reply by prash-p** [reply]
Okay, thank you!

**Reply by smistad** [reply]
I think you can do almost anything in that class. But note that the execute function runs in a separate thread. This might cause trouble for some libraries and debugging
